opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/MessagesControllerTest.java

```diff
@@ -147,30 +146,29 @@ protected TestContainerFactory getTestContainerFactory() throws TestContainerExc
         return new CustomGrizzlyTestContainerFactory();
     }
 
+    private MessagesContainer container = MessagesContainer.getInstance();
+
+    @SuppressWarnings("unchecked")
     @Before
     public void setupMessageListener() throws Exception {
-        setMessageContainer(env, new MessagesContainer());
-        env.startExpirationTimer();
+        Field messagesField = MessagesContainer.class.getDeclaredField("tagMessages");
+        messagesField.setAccessible(true);
+        ((Map<String, SortedSet<AcceptedMessage>>) messagesField.get(container)).clear();
+        container.startExpirationTimer();
```

Comment:
wouldn't it be nices to introduce public constructor (without RE) and use different instance for each test?

Reply:
like it was before (except the RE field hacking)

Reply:
IMHO you should never modify the code for the ease of testing.

Reply:
On the other hand, classes should be easily testable which is not the case when the singleton is enforced like this

Reply:
We should use dependency injection but that is for another day.

Reply:
Agreed

Reply:
You can even use the existing consumer with the RE as parameter, IMHO still better than reflection without any pain